File: week2/python/fibonacci_last_digit.py
# Uses python3
import math
def get_fibonacci_last_digit_optimized(n):
    if n <= 1:
    	return n

    previous = 0
    current = 1
    temp = 0
    for _ in range(1, n):
    	temp = current + previous
    	temp = temp%10
    	current, previous = temp, current
    return temp

if __name__ == '__main__':
    n = int(input())
    print(get_fibonacci_last_digit_optimized(n))

File: week2/C++/fibonacci_last_digit.cpp
#include <iostream>

long long get_fibonacci_last_digit_optimized(int n) {

    long long previous = 0;
    long long current  = 1;
    for (int i = 1; i < n; ++i) {
        long long temp = previous + current;
        previous = current;
        current = temp;
    }

    return current % 10;
}

int main() {
    int n;
    std::cin >> n;
    long long c = get_fibonacci_last_digit_optimized(n);
    std::cout << c << '\n';
    }
